### PR TITLE
Add platforms documentation

### PR DESCRIPTION
**Pull request by saschagrunert**
Making the platforms documentation more prominent and official by transferring the content from:

- https://github.com/kubernetes/sig-release/blob/master/release-engineering/platforms/README.md
- https://github.com/kubernetes/sig-release/blob/master/release-engineering/platforms/guide.md
- https://github.com/kubernetes/sig-release/blob/master/release-engineering/artifacts.md

I excluded the content of the archives: https://github.com/kubernetes/sig-release/blob/master/release-engineering/artifacts.md#contents-of-tar-archives

cc @kubernetes/release-engineering 

Preview: https://deploy-preview-48457--kubernetes-io-main-staging.netlify.app/releases/platforms/

### CODE REVIEW

**Comment by ameukam** [comment]
(you are too fast Sascha)

cc @kubernetes/sig-k8s-infra-leads

/lgtm

File: content/en/releases/platforms.md
Comment (windsonsea) [inline content/en/releases/platforms.md:355]: Seems [the format with `***`](https://deploy-preview-48457--kubernetes-io-main-staging.netlify.app/releases/platforms/#debs) not correct

Comment (saschagrunert) [reply content/en/releases/platforms.md]: I think we can just link to pkg.k8s.io here, done that.

File: content/en/releases/platforms.md
Comment (xmudrii) [inline content/en/releases/platforms.md:147]: Do we actually get something from running unit tests on a given architecture? I think it's mainly E2E tests and node E2E tests that we need for non-amd64 architectures

File: content/en/releases/platforms.md
Comment (xmudrii) [inline content/en/releases/platforms.md:171]: 💯 💯 💯

File: content/en/releases/platforms.md
Comment (xmudrii) [inline content/en/releases/platforms.md:208]: I personally don't like the usage of `Kubernetes trademark` here, I don't want that we get in legal stuff or eventually promise something that's not legal. It can also bring the question of if we have permission from CNCF to put it this way.

File: content/en/releases/platforms.md
Comment (xmudrii) [inline content/en/releases/platforms.md:11]: I think we should avoid usage of `we` or explicitly clarify who is `we`

File: content/en/releases/platforms.md
Comment (xmudrii) [inline content/en/releases/platforms.md:212]: The `we` comment above is specially related to this point.

File: content/en/releases/platforms.md
Comment (xmudrii) [inline content/en/releases/platforms.md:210]: Do we really need this list? I think it's very well covered by prior 4 points, and for such things, I don't think we should have tl;dr. 🙂

File: content/en/releases/platforms.md
Comment (xmudrii) [inline content/en/releases/platforms.md:238]: Given we're posting this on the public user-facing website:

- Let's replace abbreviations with full names (e.g. `k-dev` and `k/k`)
- Let's put links to each thing we refer too (again `k-dev` and `k/k`)

File: content/en/releases/platforms.md
Comment (xmudrii) [inline content/en/releases/platforms.md:252]: I would say with "string attached", i.e. that this is on the best effort basis. If Go breaks for example, there's nothing we can do, but we'll do everything in our power to accomplish this.

File: content/en/releases/platforms.md
Comment (xmudrii) [inline content/en/releases/platforms.md:329]: Is this supposed to be a heading? Also, does that following sentence even make sense?

File: content/en/releases/platforms.md
Comment (xmudrii) [inline content/en/releases/platforms.md:314]: We should provide links to those files, those are not coming with artifacts.

File: content/en/releases/platforms.md
Comment (xmudrii) [inline content/en/releases/platforms.md:327]: Let's verify this and remove the Comment prefix if it's still true. (I'm pretty sure there's no `anago` any longer)

File: content/en/releases/platforms.md
Comment (xmudrii) [inline content/en/releases/platforms.md:197]: This is not very clear and I'd consider rewording it.

**Review by xmudrii — CHANGES_REQUESTED** [review]
In addition to the comments that I left, I'm wondering if we should add and clarify that the project keeps the ultimate decision to reject a platform even if all requirements are met for any reason (e.g. cost of infra, time needed to maintain it, trust issues, etc).

**Comment by xmudrii** [comment]
Also, in addition to comments that I left, we should clarify how to start the process, e.g. by engaging with SIG Release and SIG K8s Infra (e.g. send us an email and/or attend our meetings).

File: content/en/releases/platforms.md
Comment (saschagrunert) [reply content/en/releases/platforms.md]: Sure, re phrased that part.

File: content/en/releases/platforms.md
Comment (saschagrunert) [reply content/en/releases/platforms.md]: Updated that sentence to mention the SIG Release provided tooling.

File: content/en/releases/platforms.md
Comment (saschagrunert) [reply content/en/releases/platforms.md:209]: I rephrased this to omit something like a trademark

File: content/en/releases/platforms.md
Comment (saschagrunert) [reply content/en/releases/platforms.md:211]: I agree, let's remove it.

File: content/en/releases/platforms.md
Comment (saschagrunert) [reply content/en/releases/platforms.md:235]: I rephrased that to include the best effort note.

File: content/en/releases/platforms.md
Comment (saschagrunert) [reply content/en/releases/platforms.md:312]: I'm removing this section completely, no need to mention that they're "shipped".

File: content/en/releases/platforms.md
Comment (saschagrunert) [reply content/en/releases/platforms.md:297]: Removing that section completely.

File: content/en/releases/platforms.md
Comment (saschagrunert) [reply content/en/releases/platforms.md:310]: I removed that section entirely.

File: content/en/releases/platforms.md
Comment (saschagrunert) [reply content/en/releases/platforms.md]: I agree, I rephrased the section to make the stance more clear and precise.

**Comment by saschagrunert** [comment]
> In addition to the comments that I left, I'm wondering if we should add and clarify that the project keeps the ultimate decision to reject a platform even if all requirements are met for any reason (e.g. cost of infra, time needed to maintain it, trust issues, etc).

Adding that note in a dedicated section "Step 0: Engage with the community".

> Also, in addition to comments that I left, we should clarify how to start the process, e.g. by engaging with SIG Release and SIG K8s Infra (e.g. send us an email and/or attend our meetings).

Same for this one.

File: content/en/releases/platforms.md
Comment (upodroid) [reply content/en/releases/platforms.md:164]: Can we add the following as a list?
- The arch must have official go builds
- The arch must be supported by Debian and included in LTS.
- A list of go/apps that must be built for the arch.

File: content/en/releases/platforms.md
Comment (upodroid) [inline content/en/releases/platforms.md:272]: Can we explore deprecating these GCE files? Is there an issue for this?

Comment (BenTheElder) [reply content/en/releases/platforms.md:272]: they're primarily used by kube-up, it will be a little painful though, various places expect to be able to download the tarball.

we have rarely removed artifacts, even the bash scripts

File: content/en/releases/platforms.md
Comment (BenTheElder) [reply content/en/releases/platforms.md:164]: > Do we actually get something from running unit tests on a given architecture? I think it's mainly E2E tests and node E2E tests that we need for non-amd64 architectures

it gives us higher confidence that we don't have platform specific integer sizes and issues like that, and that all the code compiles. it's a low baseline, not sufficient on it's own, but if we can't even get unit test signal then ...

File: content/en/releases/platforms.md
Comment (BenTheElder) [inline content/en/releases/platforms.md:77]: we also have arm64-darwin, don't we?

File: content/en/releases/platforms.md
Comment (marosset) [inline content/en/releases/platforms.md:82]: Is there a reason why windows isn't `Tier 1`?

I believe all of the requirements are met.

* for build artifacts, all of the components needed to join windows nodes to the cluster (kubelet, kubeadm, kubectl) are produced as part of official releases.
* We have CI test up that is monitored by the release team (https://testgrid.k8s.io/sig-release-1.31-informing#capz-windows-1.31)
* We have a lot of documentation on k8s.io w.r.t. managing Windows nodes in a cluster

File: content/en/releases/platforms.md
Comment (saschagrunert) [reply content/en/releases/platforms.md:272]: I'm not exactly sure where we publish those files.

File: content/en/releases/platforms.md
Comment (saschagrunert) [reply content/en/releases/platforms.md:77]: True, added that.

File: content/en/releases/platforms.md
Comment (saschagrunert) [reply content/en/releases/platforms.md:83]: Hm, right now we state:

> For Tier 1 only the `blocking` jobs are required, the `informing` ones are optional

Which technically would mean that we need to have all the jobs available in blocking for linux/amd64 also for windows. Would that make sense? Maybe we should think about creating a tier-based dashboard view. (cc @kubernetes/release-managers)

Comment (ameukam) [reply content/en/releases/platforms.md:83]: > all of the components needed to join windows nodes to the cluster (kubelet, kubeadm, kubectl) are produced as part of official releases.

This is not completely true. We still don't produce binaries kube-apiserver, kube-controller-manager, kube-scheduler capable to run to Windows environments.

**Review by sftim — CHANGES_REQUESTED** [review]
Could we document this within https://k8s.dev/, and then have https://k8s.io/releases/ link there?

(I recommend doing that; in fact I am going to make this a **Request changes** review to convey my strength of feeling).

---

We already list what platforms we make artefacts available for. If we want to list the tier as well, that information belongs in https://kubernetes.io/releases/download/
 
Right now we list all architectures as “supported”, end of.

**Comment by sftim** [comment]
> if we should add and clarify that the project keeps the ultimate decision to reject a platform even if all requirements are met for any reason (e.g. cost of infra, time needed to maintain it, trust issues, etc)

Do we commit to continue supporting a platform for at least _n_ days / minor releases / whatever from the moment we announce the plan to discontinue support?

File: content/en/releases/platforms.md
Comment (marosset) [reply content/en/releases/platforms.md:83]: So is it a requirement that we be able to build / run / test the control plane components for a given platform for it to be `Tier 1` supported?

Also, those components (kube-apiserver, kube-controller-manager, kube-schedule) aren't required to join a node to an existing cluster. We do build / test all of the the componenets required to join a node to the cluster, just not to have to run control-plane components on those nodes.

**Comment by saschagrunert** [comment]
> Could we document this within https://k8s.dev/, and then have https://k8s.io/releases/ link there?
> 
> (I recommend doing that; in fact I am going to make this a **Request changes** review to convey my strength of feeling).

I'm unsure about this, how do @kubernetes/release-engineering folks think about it?

File: content/en/releases/platforms.md
Comment (sftim) [inline content/en/releases/platforms.md:182]: Is this right for end-user docs?

File: content/en/releases/platforms.md
Comment (sftim) [inline content/en/releases/platforms.md:230]: Are we implying that end users should join this mailing lsit?

File: content/en/releases/platforms.md
Comment (saschagrunert) [reply content/en/releases/platforms.md:208]: Fixed up that section and removed the "we".

File: content/en/releases/platforms.md
Comment (saschagrunert) [reply content/en/releases/platforms.md:229]: A this should be the announce mailing list.

File: content/en/releases/platforms.md
Comment (sftim) [inline content/en/releases/platforms.md:48]:
```suggestion
Tier 2 platforms are considered as _expected to build_. To achieve this, they
```

File: content/en/releases/platforms.md
Comment (sftim) [inline content/en/releases/platforms.md:55]: Can there ever be 2 unavailable features for a platform?

**Comment by sftim** [comment]
Because it's much better to have some docs than no docs, about this detail:

/lgtm

**Comment by saschagrunert** [comment]
We discussed that during our SIG Release meeting and will start a new draft following a possible KEP.

**Comment by justaugustus** [comment]
@kubernetes/sig-release-leads agree to hold this documentation until we have a clear decision on how to communicate the support tiers. We'll take this back to a Google Doc and follow up here and in a KEP once ready.

**Pull request closed without merging**
